src/duty/_completion.py

```diff
@@ -0,0 +1,41 @@
+from typing import Optional
+
+CompletionCandidateType = tuple[str, Optional[str]]
+
+
+class CompletionParser:
+    @classmethod
+    def parse(cls, candidates: list[CompletionCandidateType], shell: str) -> str:
+        """Parses a list of completion candidates for selected shell completion command.
+
+        Parameters:
+            candidates: List of completion candidates with optional descriptions.
+            shell: Shell for which to parse the candidates.
+
+        Raises:
+            NotImplementedError: When parser is not implemented for selected shell.
+
+        Returns:
+            String to be passed to shell completion command.
+        """
+        try:
+            return getattr(cls, f"_{shell}")(candidates)
+        except AttributeError as exc:
+            msg = f"CompletionParser method for {shell!r} shell is not implemented!"
+            raise NotImplementedError(msg) from exc
+
+    @staticmethod
+    def _zsh(candidates: list[CompletionCandidateType]) -> str:
+        def parse_candidate(item: CompletionCandidateType) -> str:
+            completion, help_text = item
+            # We only have space for one line of description,
+            # so I remove descriptions of sub-command parameters from help_text
+            # by removing everything after the first newline.
+            # I don't think it is the best approach and should be discussed.
+            return f"{completion}: {help_text or '-'}".split("\n", 1)[0]
```

Comment:
- functions docstrings: good, it's reasonable to keep only the first line, as docstrings' first line should be a complete sentence anyway.
- parameter docstrings: we don't parse them yet anyway (from looking at code below). When we do (using Griffe?), I don't think there's a more robust way than keeping the first line anyway. We don't want to enter natural language processing territory (splitting sentences is hard). Maybe we could recommend to use single-line descriptions for parameters, or at least to have a complete sentence as first line.